src/v/pandaproxy/test/sharded_client_cache.cc

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "pandaproxy/sharded_client_cache.h"
+
+#include "config/node_config.h"
+#include "config/rest_authn_endpoint.h"
+#include "kafka/client/configuration.h"
+#include "pandaproxy/types.h"
+
+#include <seastar/core/sleep.hh>
+#include <seastar/testing/thread_test_case.hh>
+#include <seastar/util/defer.hh>
+
+#include <boost/test/unit_test.hpp>
+
+#include <chrono>
+
+namespace pp = pandaproxy;
+
+SEASTAR_THREAD_TEST_CASE(test_sharded_client_cache) {
```

Comment:
if we aren't going to test the non-sharded version separately, then it is probably reasonable to combine them.